octoprint_simple_filament_change_buttons/static/js/simple_filament_change_buttons.js

```diff
@@ -11,29 +11,70 @@ $(function() {
 		// assign the injected parameters
 		self.controlViewModel = parameters[0];
 
+		self.unloadFilament = function() {
+			cmds = [
+				// Heat to specified temp
+				"M109 S" + self.temperature(),
+				// Do unload
+				"M702",
+				// Cool hotend
+				"M104 S0"
+			]	
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.loadFilament = function() {
+			cmds = [
+				// Heat to specified temp
+				"M109 S" + self.temperature(),
+				// Do load
+				"M701",
+				// Cool hotend
+				"M104 S0"
+			]	
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.changeFilament = function() {
+			cmds = [
+				// Change filament at specified temp
+				"M600 R" + self.temperature(),
+			]		
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.resume = function() {
+			cmds = [
+				'M108'
+			]		
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.temperature = function()
+		{
+			// The temperature input field has a "placeholder" attribute that is equal to it's name
+			return $('*[placeholder = "Temperature"]')[0].value
+		}
+
+
 		self.getAdditionalControls = function() {
 			return [{
 				'layout': 'horizontal', 'name': 'Filament Change', 'children':[
 					/*{'commands': [
 						'M125'
 						],
 						'additionalClasses': 'sfcb fa-pause', 'name': ' Park'},*/
-					{'commands': [
-						'M702'
-						],
-						'additionalClasses': 'sfcb fa-fast-backward', 'name': ' Unload'},
-					{'commands': [
-						'M701'
-						],
-						'additionalClasses': 'sfcb fa-step-forward', 'name': ' Load'},
-					{'commands': [
-						'M600'
-						],
-						'additionalClasses': 'sfcb fa-sync', 'name': ' Change Filament'},
-					{'commands': [
-						'M108'
-						],
-						'additionalClasses': 'sfcb fa-play', 'name': ' Resume'},
+
+					{name: ' Unload', javascript: self.unloadFilament, additionalClasses: 'sfcb fa-fast-backward'},
+					{name: ' Load', javascript: self.loadFilament, additionalClasses: 'sfcb fa-step-forward'},
+					{name: ' Change Filament', javascript: self.changeFilament, additionalClasses: 'sfcb fa-sync'},
+					{name: ' Resume', javascript: self.resume, additionalClasses: 'sfcb fa-play'},
+
+					{'input': [{
+							name: 'Temperature',
+							default: 200
+						}]
```

Comment:
Will this be persisted between loads of the UI?
Maybe this should be a setting instead?

Reply:
Could possibly be a setting, though I was trying to avoid making too many changes - I'm not sure that will be possible while keeping it as SimpleControlPlugin